Tangem/Modules/Swapping/SwappingViewModel.swift

```diff
@@ -210,31 +210,27 @@ private extension SwappingViewModel {
 
     func openPermissionView() {
         let state = exchangeManager.getAvailabilityState()
-        let source = exchangeManager.getExchangeItems().source
 
         guard case .available(let result, let info) = state,
               result.isPermissionRequired,
-              fiatRatesProvider.hasRates(for: source) else {
+              fiatRatesProvider.hasRates(for: info.sourceCurrency) else {
             // If we don't have enough data disable button and refresh()
             mainButtonIsEnabled = false
             exchangeManager.refresh(type: .full)
 
             return
         }
 
-        runTask(in: self) { obj in
-            let fiatFee = try await obj.fiatRatesProvider.getFiat(for: source, amount: info.fee)
-            let inputModel = SwappingPermissionInputModel(
-                fiatFee: fiatFee,
-                transactionInfo: info
-            )
-
-            obj.stopTimer()
-
-            await runOnMain {
-                obj.coordinator.presentPermissionView(
-                    inputModel: inputModel,
-                    transactionSender: obj.transactionSender
+        Task {
+            let fiatFee = try await fiatRatesProvider.getFiat(for: info.sourceBlockchain, amount: result.fee)
```

Comment:
в принципе хватило было сделать
```suggestion
            let fiatFee = try await fiatRatesProvider.getFiat(for: source.blockchain, amount: result.fee)
```
А все остальное оставить как было)

Reply:
Да короче можешь вот сделать, и все, только одну строчку, я тут посчитал комиссию не в валюте сети, а в валюте токена, вот и косяк)